qqq-backend-module-rdbms/src/main/java/com/kingsrook/qqq/backend/module/rdbms/actions/RDBMSInsertAction.java

```diff
@@ -146,16 +146,16 @@ public InsertOutput execute(InsertInput insertInput) throws QException
             // todo sql customization - can edit sql and/or param list
             // todo - non-serial-id style tables
             // todo - other generated values, e.g., createDate...  maybe need to re-select?
-            List<Integer> idList = QueryManager.executeInsertForGeneratedIds(connection, sql.toString(), params);
-            int           index  = 0;
+            List<Serializable> idList = QueryManager.executeInsertForGeneratedIds(connection, sql.toString(), params, table.getFields());
```

Comment:
i think rather than passing the fields map, and having the other method look for one named "id", this would be better as:
```
table.getField(table.getPrimaryKeyField().getType())
```